odd-platform-ui/src/components/shared/AppSelect/AppSelect.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { TextFieldProps } from '@mui/material';
+import {
+  StyledAppSelect,
+  AppSelectSizes,
+} from 'components/shared/AppSelect/AppSelectStyles';
+import DropdownIcon from 'components/shared/Icons/DropdownIcon';
+
+interface AppSelectProps
+  extends Pick<
+    TextFieldProps,
+    | 'onClick'
+    | 'onChange'
+    | 'sx'
+    | 'value'
+    | 'label'
+    | 'defaultValue'
+    | 'type'
+    | 'name'
+    | 'fullWidth'
+    | 'select'
+    | 'SelectProps'
+    | 'onSelect'
+    | 'id'
+  > {
+  size?: AppSelectSizes;
+  selectNative?: boolean;
+}
+const AppSelect: React.FC<AppSelectProps> = ({
+  children,
+  size = 'medium',
+  onClick,
+  sx,
+  value,
+  onChange,
+  label,
+  defaultValue,
+  type,
+  fullWidth = true,
+  select,
+  SelectProps,
+  onSelect,
+  id,
+  selectNative,
+}) => (
+  <StyledAppSelect
+    $size={size}
+    $isLabeled={!!label}
+    variant="outlined"
+    fullWidth={fullWidth}
+    sx={sx}
+    value={value}
+    onChange={onChange}
+    onClick={onClick}
+    label={label}
+    defaultValue={defaultValue}
+    type={type}
+    select={select || selectNative}
```

Comment:
You are created a component `AppSelect`, but in all places where you use it - you must pass prop `select`. It's strange, when you said AppSelect that it should work like select.
In most cases we use simple select (not native), so I think that it would be better for you to make the behavior of the component as "Select" by default, and where you need to use "selectNative" - enable it through the prop.